### ISSUE TITLE
Description of splits + some kNN questions

### ISSUE THREAD

**Issue by tsirif** [Annusha/temperature_schedules#3]
The repository contains some splits on datasets. Can you provide with a bit more information about those?

For example, I guess `i` stands for imbalanced and `b` for balanced. Also, I guess split1,2,3,4,5 correspond to 5 randomized splits from the same dataset. I don’t know what `D` and `S` are for. I guess the imbalance ratio is 100 as stated in the paper.

Some further reproducibility questions:
1. Regarding the KNN experiments, did you collect statistics on the balanced or imbalanced training set?
2. The oracle experiments presented in figure 1 are super interesting! what was the number of neighbors used in that case? Also, same to the question above.
3. For kNN, to you use embeddings from the hypersphere or from the outputs of the backbone model (before the projector)?

**Comment by Annusha** [comment]
Hi
This repo borrows splits from [SDCLR](https://github.com/VITA-Group/SDCLR). In this work, we do not use all these scripts, only a subset (training imbalanced D splits), as they represent also different tasks. We didn't clean this part of the code, but it should work with any of the subsets.  
For training, we use only D scripts to be directly comparable with SDCLR. S stands for few-shot evaluation in SDCLR. Our evaluation protocol is slightly different from SDCLR, and you can find it e.g. [here](https://github.com/Annusha/temperature_schedules/blob/main/notebooks/simclr_tau_cifar10.ipynb). We reproduce (most of) SDCLR numbers and run their pretrained models in our evaluation scripts. For different imbalance ratios, we use the standard practice of subsampling e.g. [here](https://github.com/kaidic/LDAM-DRW/blob/master/imbalance_cifar.py). As we use it only with MoCo, the code is yet not ready for release.
Also, originally, long-tail papers use only one split per imbalance ratio (alphabetically ordered) for training, whereas SDCLR introduces several splits but only for imbalance ratio 100. In our work, we use only three splits for both us and SDCLR for imb.ratio 100. 

1. KNN accuracy is tested with an imbalanced train split with 1 and 10 neighbours
2. The results for K=1, the results hold for larger K as well
3. We use normalized outputs of the backbone. Note that it's different for the original SDCLR, they do not normalize before linear classification which gives the same or slightly worse results than our reproduction with normalization.

**Comment by tsirif** [comment]
Thank you for the responses!

**Comment by tsirif** [comment]
Thank you for the responses!
